### DISCUSSION TITLE
Setting gradient background color with local rendering

### DISCUSSION THREAD

**Discussion by bigfooted** [Kitware/trame#256] [Q&A] [answered]
Hi all,
I have modified the simple cone with [LocalRendering.py](https://github.com/Kitware/trame/blob/master/examples/06_vtk/01_SimpleCone/LocalRendering.py) and added a gradient background:
 
```
renderer.SetBackground(0.9,0.9,1.0)
renderer.SetBackground2(0.0, 0.0, 0.0)
renderer.GradientBackgroundOn()
renderer.ResetCamera()
#renderWindow.Render()
```

But this only works when I switch from local to remote rendering (in local, only the color from SetBackground is used):
```
    view = vtk.VtkRemoteView(renderWindow)
    #view = vtk.VtkLocalView(renderWindow)
    ctrl.view_update = view.update
    ctrl.view_reset_camera = view.reset_camera
```

(and the only other thing I had to add was `import vtkmodules.vtkRenderingOpenGL2  # noqa`)

So what am I doing wrong? :-)

**Comment by jourdain** [comment, marked as answer]
Nothing on your side, it is just that vtk.js don't do gradient on the GL side. To do it you have to use a transparent background and do your gradient on the CSS side.
